### PR TITLE
[Brave News]: Don't pick articles with no image as Hero card

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/34595

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by fallaciousreasoning** [comment]
**Note:** This isn't going to look good until the Nala tokens update lands with the new Inter font

File: components/brave_news/browser/feed_v2_builder.cc [hunk 1]
```diff
@@ -327,13 +329,20 @@ std::vector<mojom::FeedItemV2Ptr> GenerateBlock(
...
+      });
 
-  result.push_back(mojom::FeedItemV2::NewHero(
-      mojom::HeroArticle::New(std::move(hero_article))));
+  // We might not be able to generate a hero card, if none of the articles in
+  // this feed have an image.
+  if (hero_article) {
+    result.push_back(mojom::FeedItemV2::NewHero(
+        mojom::HeroArticle::New(std::move(hero_article))));
+  }
```
Comment (sangwoo108) [inline components/brave_news/browser/feed_v2_builder.cc:345]: Just double checking: we were early returning when we don't have hero article before. Now code from L387 will be executed regardless of hero article. This is intended, isn't it?

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: components/brave_news/browser/feed_v2_builder.cc (hunk 1 above)
Comment (sangwoo108) [reply components/brave_news/browser/feed_v2_builder.cc:345]: Thanks! That makes sense

**Comment by fallaciousreasoning** [comment]
@sangwoo108 I've dropped the font changes from this PR - they're coming in https://github.com/brave/brave-core/pull/21295

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.63.36 Chromium: 120.0.6099.71 (Official Build) nightly (x86_64)
-- | --
Revision | 8dc787c0e2867326da0f341f6991bb4334e0c4f4
OS | macOS Version 14.3 (Build 23D5033f)
```


### First, reproduced easily using `1.62.105`

example | example | example
---------|----------|---------
Screenshot 2023-12-12 at 4 11 28 PM | Screenshot 2023-12-12 at 4 11 47 PM | Screenshot 2023-12-12 at 4 11 53 PM

**Pull request merged**